### PR TITLE
ELEMENTS-1641: FORWARD PORT: Use feature flag for view vs download

### PR DESCRIPTION
**Pull request by rahuljain-dev**
https://jira.nuxeo.com/browse/ELEMENTS-1641
(FORWARD PORT)

### CODE REVIEW

File: core/nuxeo-document.js [hunk 1]
```diff
@@ -182,12 +182,25 @@ import './nuxeo-resource.js';
 
       this.$.nxResource.addEventListener('response', (e) => {
         this.documentData = e.detail && e.detail.response ? e.detail.response : null;
-        if (this.documentData) {
-          this.setDocumentViewDownloadProp();
+        const documentDataProperties = this.documentData && this.documentData.properties;
+        const fileContent = documentDataProperties && documentDataProperties['file:content'];
+        const files = documentDataProperties && documentDataProperties['files:files'];
```
Comment (rakeshkumar1019) [inline core/nuxeo-document.js:187]: Can we use dynamic xpath `file:content` & `files:files` here.

File: core/nuxeo-document.js [hunk 2]
```diff
@@ -182,12 +182,25 @@ import './nuxeo-resource.js';
...
+        if (
+          ((fileContent && fileContent.data) ||
+            (files && files.length > 0 && files.every((item) => item.file && item.file.data))) &&
```
Comment (rakeshkumar1019) [inline core/nuxeo-document.js:190]: **IMP**: `files && files.length > 0` we can do both checks using this `Array.isArray(files)`.

**Review by poonamyadav252 — APPROVED** [review]

File: core/nuxeo-document.js (hunk 1 above)
Comment (rahuljain-dev) [reply core/nuxeo-document.js:187]: > Can we use dynamic xpath `file:content` & `files:files` here.

This is in reference to existing code used in the application. We can take it as an improvement in near future for all the places used

File: core/nuxeo-document.js (hunk 2 above)
Comment (rahuljain-dev) [reply core/nuxeo-document.js:190]: > **IMP**: `files && files.length > 0` we can do both checks using this `Array.isArray(files)`.

This is in reference to existing code used in the application. We can take it as an improvement in near future for all the places used

**Review by rakeshkumar1019 — APPROVED** [review]

**Review by rajatgoel-gl — APPROVED** [review]

**Pull request merged**